doc/bgp_loading_optimization/bgp-loading-optimization-hld.md

```diff
@@ -179,35 +177,16 @@ Once `orchagent` `doTask` writes data to ASIC_DB, it waits for response from `sy
-While the producer has APIs associated with lua scripts for Redis operations, each script ends with a _PUBLISH_ command to notify the downstream consumers.
+Orchagent subscribe data with redis based ConsumerStateTable, sonic-swss-common has ZMQ based ZmqProducerStateTable/ZmqConsumerStateTable, here are compare with redis based tables:
+1. 200+ times faster than redis based table, ZMQ table can transfer 800K route entry per-second, redis table can only transfer 3.7K route entry per-second.
+2. Fully compatible with ProducerStateTable/ConsumerStateTable.
+3. Asynchronous update redis DB with background thread.
 
-Since we have employed Redis pipeline to queue commands up and flush them in a batch, it's unnecessary to _PUBLISH_ for each command. We can attach a _PUBLISH_ at the end of the command queue when the pipeline flushes, then the whole batch could share this single _PUBLISH_ and we reduce traffic for O(n) _PUBLISH_ to O(1).
 <figure align=center>
-    <img src="images/BatchPub.png" height=auto>
-</figure>  
-
-#### 1.2 Reduce pipeline flush frequency
-
-> Redis pipeline flushes itself when it's full, otherwise it temporarily holds the redis commands in its buffer.  
-The commands would not get stuck in the pipeline since Fpmsyncd would also flush the pipeline, and this behavior is event-triggered with _select_ method.
-
-Firstly, we could increase pipeline buffer size from the default 125 to 10k, which would decrease the frequency of the pipeline flushing itself.  
-Secondly, we could skip Fpmsyncd flushes when it's not that long since the last flush and set a _flush timeout_ to determine the threshold.  
-To avoid commands lingering in the pipeline due to skip, we change the _select timeout_ of Fpmsyncd from _infinity_ to _flush timeout_ after a successful skip to make sure that these commands are eventually flushed. And after flushing the lingered commands, the _select timeout_ of Fpmsyncd would change back to _infinity_ again.  
-To make sure that consumers are able to get all the modified data when the number of _PUBLISH_ is equal to the number of flushes, while still keep the consumer pop size as 125, consumer needs to do multiple pops for a single _PUBLISH_.  
-If there is a batch of fewer than 10 routes coming to the pipeline, they would be directly flushed, in case they are important routes.
-
-#### 1.3 Discard the state table with prefix _
->
-> Pipeline flushes data into the state table and use the data structure _KeySet_ to mark modified keys.  
-When Orchagent is notified of new data coming, it recognized modified keys by _KeySet_, then transfers data from the state table to the stable table, then deletes the state table.
-
-We propose to discard state tables, directly flush data to stable tables, while keep the data structure _KeySet_ to track modified keys.
+    <img src="images/zmq-diagram.png" height=auto>
+</figure>
 
 ### 2. Add an assistant thread to the monolithic Orchagent/Syncd main event loop workflow
 
```

Comment:
This part need re-write with Syncd code example, because orchagent already improve with ZMQ.